cooked-validators/src/Cooked/MockChain/Direct.hs

```diff
@@ -67,16 +67,19 @@ mcstToUtxoState MockChainSt {mcstIndex, mcstDatums} =
     . utxoIndexToTxOutMap
     $ mcstIndex
   where
-    go :: (Pl.TxOutRef, PV2.TxOut) -> Maybe (Pl.Address, UtxoValueSet)
-    go (txOutRef, PV2.TxOut {PV2.txOutAddress, PV2.txOutValue, PV2.txOutDatum}) =
-      case txOutDatum of
-        Pl.NoOutputDatum -> return (txOutAddress, UtxoValueSet [(txOutValue, TxSkelOutNoDatum)])
-        Pl.OutputDatum datum -> do
-          txSkelOutDatum <- Map.lookup (Pl.datumHash datum) mcstDatums
-          return (txOutAddress, UtxoValueSet [(txOutValue, txSkelOutDatum)])
-        Pl.OutputDatumHash hash -> do
-          txSkelOutDatum <- Map.lookup hash mcstDatums
-          return (txOutAddress, UtxoValueSet [(txOutValue, txSkelOutDatum)])
+    go :: (Pl.TxOutRef, PV2.TxOut) -> Maybe (Pl.Address, UtxoPayloadSet)
+    go (txOutRef, out@PV2.TxOut {PV2.txOutAddress, PV2.txOutValue, PV2.txOutDatum}) =
```

Comment:
This surely comes from the previous version of the code as well, but `go` is typically used for recursive "inner loop" functions. Should this better be named like `extractUtxoPayload` or something like that?

Reply:
I agree, this was misleading and `extractUtxoPayload` is way more meaningful.